phpseclib/Crypt/Common/SymmetricKey.php

```diff
@@ -2006,17 +2015,17 @@ private function setupMcrypt()
     protected function pad($text)
     {
         $length = strlen($text);
-
+        if ($length % $this->block_size == 0) {
+            return $text;
+        }
         if (!$this->padding) {
-            if ($length % $this->block_size == 0) {
-                return $text;
-            } else {
-                throw new \LengthException("The plaintext's length ($length) is not a multiple of the block size ({$this->block_size}). Try enabling padding.");
-            }
+            throw new \LengthException("The plaintext's length ($length) is not a multiple of the block size ({$this->block_size}). Try enabling padding.");
         }
 
         $pad = $this->block_size - ($length % $this->block_size);
-
+        if ($this->custom_padding_str != null){
+            return str_pad($text, $length + $pad, $this->custom_padding_str);
```

Comment:
So what if you have padding enabled and are using a custom padding string and want to decrypt some text? phpseclib will try to unpad using the PKCS7 technique and it'll very possibly fail depending on what the custom padding string is.

If you want to employ custom padding I'd suggest doing it outside of the the SymmetricCipher instance. SSH2.php does this. It has the packet length, the padding length, the data and then random padded data. ie. it doesn't have the SymmetricCipher instance do the padding - it does it itself. Because the padding requires a plaintext that is formatted in a certain way - a format that isn't shared by other protocols or formats or whatever.